src/pages/EmergencyViewPage/EmergencyViewPage.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { addDays } from "date-fns";
+import dayjs from "dayjs";
+import advancedFormat from "dayjs/plugin/advancedFormat";
+
+import {
+  DEFAULT_VENUE_BANNER,
+  DEFAULT_VENUE_LOGO,
+  IFRAME_ALLOW,
+} from "settings";
+
+import {
+  eventTimeAndOrderComparator,
+  isEventLive,
+  isEventWithinDateAndNotFinished,
+} from "utils/event";
+import { range } from "utils/range";
+import { currentVenueSelectorData } from "utils/selectors";
+import { formatDateRelativeToNow } from "utils/time";
+
+import useConnectCurrentVenue from "hooks/useConnectCurrentVenue";
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+import useVenueScheduleEvents from "hooks/useVenueScheduleEvents";
+
+import { updateTheme } from "pages/VenuePage/helpers";
+
+import WithNavigationBar from "components/organisms/WithNavigationBar";
+
+import { LoadingPage } from "components/molecules/LoadingPage";
+import { ScheduleEventSubListNG } from "components/molecules/ScheduleEventListNG/ScheduleEventSubListNG";
+
+import EmergencyViewRoom from "./EmergencyViewRoom";
+
+import "./EmergencyViewPage.scss";
+
+export const emptyPersonalizedSchedule = {};
+
+const tabs = [
+  { id: 0, title: "Spaces" },
+  { id: 1, title: `What's on` },
+];
+
+export const EmergencyViewPage: React.FC = () => {
+  const [selectedTab, updateTab] = useState(0);
+  const venueId = useVenueId() || "";
+  useConnectCurrentVenue();
+
+  const venue = useSelector(currentVenueSelectorData);
+  const { userWithId } = useUser();
+  const userEventIds =
+    userWithId?.myPersonalizedSchedule ?? emptyPersonalizedSchedule;
+
+  const {
+    descendantVenues,
+    dayDifference,
+    liveAndFutureEvents,
+    firstScheduleDate,
+  } = useVenueScheduleEvents({
+    venueId,
+    userEventIds,
+  });
+
+  const venueRequestStatus = useSelector(
+    (state) => state.firestore.status.requested.currentVenue
+  );
+  const redirectUrl = venue?.config?.redirectUrl ?? "";
+  const { hostname } = window.location;
+
+  useEffect(() => {
+    if (redirectUrl && redirectUrl !== hostname) {
+      window.location.hostname = redirectUrl;
+    }
+  }, [hostname, redirectUrl]);
+
+  dayjs.extend(advancedFormat);
+
+  useEffect(() => {
+    if (!venue) return;
+
+    // @debt replace this with useCss?
+    updateTheme(venue);
+  }, [venue]);
```

Comment:
This is repeated in a codebase. It needs to be extracted into a separate hook everywhere and not duplicated once more here. And maybe replaced with `useCss`:)